src/metabase/models/params/chain_filter.clj

```diff
@@ -560,7 +569,9 @@
      (-> (unremapped-chain-filter field-id constraints options)
          (update :values add-human-readable-values v->human-readable))
 
-     (and (use-cached-field-values? field-id) (nil? @the-remapped-field-id))
+     (and (use-cached-field-values? field-id)
+          (nil? @the-remapped-field-id)
+          (check-field-value-query-permissions! field-id constraints options))
      (cached-field-values field-id constraints options)
```

Comment:
I think the way this code is structured sort of suggests that `check-field-value-query-permissions` can return falsey in some situations, when in reality the expectation is that it's meant to throw an exception if the check failures, and the return value doesn't matter. I think changing the code to this is more in line where how we do this sort of thing elsewhere and is clearer

```suggestion
          (nil? @the-remapped-field-id))
     (do
       (check-field-value-query-permissions field-id constraints options)
       (cached-field-values field-id constraints options))
```

Reply:
if you do this, then you don't need to have `check-field-value-query-permissions` return `true` (which seems a little icky tbh), you can just have it return nothing